pkgs/development/libraries/libdatachannel/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, ninja
+, pkg-config
+, libnice
+, nlohmann_json
+, openssl
+, srtp
+, usrsctp
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libdatachannel";
+  version = "0.18.4";
+
+  src = fetchFromGitHub {
+    owner = "paullouisageneau";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-kSK+5gFMG6tq89R1m08gNBKPdwyR/mLEDhWXQ/uk34o=";
+    fetchSubmodules = true;
+  };
+
+  outputs = [ "out" "dev" ];
+
+  strictDeps = true;
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+  ];
+  buildInputs = [
+    libnice
+    nlohmann_json
+    openssl
+    srtp
+    usrsctp
+  ];
+
+  cmakeFlags = [
+    "-DUSE_NICE=ON"
+    "-DUSE_SYSTEM_JSON=ON"
+    "-DUSE_SYSTEM_SRTP=ON"
+    "-DUSE_SYSTEM_USRSCTP=ON"
+  ];
+
```

Comment:
```suggestion
  postPatch = ''
    # TODO: Remove when updating to 0.19.x, and add
    # -DUSE_SYSTEM_USRSCTP=ON and -DUSE_SYSTEM_PLOG=ON to cmakeFlags instead
    mkdir -p deps/{usrsctp,plog}
    cp -r --no-preserve=mode ${srcOnly usrsctp}/. deps/usrsctp
    cp -r --no-preserve=mode ${srcOnly plog}/. deps/plog
  '';
```
That allows us to use the nixpkgs versions of usrsctp and plog. The advantage is that if we update usrsctp in nixpkgs, then libdatachannel will also get that updated version.

<details>
<summary>plog derivation scaffold (you can add me to maintainers if desired)</summary>

```
{ lib
, stdenv
, fetchFromGitHub
, cmake
}:

stdenv.mkDerivation rec {
  pname = "plog";
  version = "1.1.9";

  outputs = [ "out" "dev" ];

  src = fetchFromGitHub {
    owner = "SergiusTheBest";
    repo = pname;
    rev = version;
    hash = "sha256-CARTr1EEqXNJtp6XwHhr7aiRBiYR0cClzexpNCMaQJc=";
  };

  nativeBuildInputs = [
    cmake
  ];

  cmakeFlags = [
    "-DPLOG_BUILD_SAMPLES=NO"
  ];

  meta = with lib; {
    description = "Portable, simple and extensible C++ logging library";
    homepage = "https://github.com/SergiusTheBest/plog";
    license = licenses.mit;
    platforms = platforms.linux ++ platforms.darwin;
    maintainers = with maintainers; [ __TODO__ ];
  };
}
```

</details>

Reply:
> The advantage is that if we update usrsctp in nixpkgs, then libdatachannel will also get that updated version.

There's no new releases for usrsctp, so that's not a good idea.

Reply:
Thanks, I added a derivation with libdatachannel usrsctp version.